core/src/main/resources/schema/infinispan-config-10.1.xsd

```diff
@@ -15,6 +15,8 @@
     </xs:annotation>
   </xs:element>
 
+  <xs:element name="cache-container" type="tns:cache-container"/>
```

Comment:
Will it be both a top level element and an element inside <infinispan> ? What about the effect of this on builders, attributes, parsers, XML and JSON formats? Maybe some tests should be added

Reply:
It's only to allow us to add the schema to xincluded bits (see https://github.com/infinispan/infinispan/pull/7716/files#diff-2f49a6c5217bf1c181ca6354298c0c8dR1)
It doesn't change the type of files we can parse. The change I've made in the second commit test this.

Reply:
My understanding is that a file with simply the ```<cache-container>``` as the top level element will be valid whereas today is not possible?

Reply:
No, it will still be invalid. But I have not found another way to allow validation in xincluded fragments without making the elements top-level. It is merely a convenience for adding schema.
I guess the only way is to have a separate schema to be used in included fragments which imports the infinispan schema and adds those "sub" top-level elements without affecting the main XSD.

Reply:
Will it be invalid according to the XSD or according to the internal parser, or both?

Reply:
@gustavonalle I have updated the schema as discussed on Zulip:
* the infinispan-config.xsd is unchanged and only allows the `infinispan` top level element
* xincluded fragments must use infinispan-config-fragment.xsd which allow other top-level elements (for practical reasons I've only included the most useful, but it can be expanded at any time).